nntrainer/graph/network_graph.cpp

```diff
@@ -702,14 +701,11 @@ int NetworkGraph::initialize(std::shared_ptr<Manager> manager) {
       for (unsigned int i = 0; i < input_layers.size(); ++i) {
         auto in_layer_node = getLayerNode(input_layers[i]);
 
-        unsigned int location = 0;
-        for (unsigned int j = 0; j < in_layer_node->getNumOutputConnections();
-             ++j) {
-          if (in_layer_node->getOutputLayers()[j] == lnode->getName()) {
-            location = j;
-            break;
-          }
-        }
+        auto const &in_layer_out_connect = in_layer_node->getOutputLayers();
+        unsigned int location =
+          std::find(in_layer_out_connect.begin(), in_layer_out_connect.end(),
+                    lnode->getName()) -
```

Comment:
how about adding logic to check if the layer is found?

Reply:
Will add this check in DEBUG mode.

Reply:
done.